sleap/nn/utils.py

```diff
@@ -148,30 +148,15 @@ def reset_input_layer(
         new_shape = (None, None, None, keras_model.input_shape[-1])
 
     model_config = keras_model.get_config()
-
-    input_layer_name = model_config["layers"][0]["name"]
-    model_config["layers"][0] = {
-        "name": f"{input_layer_name}",
-        "class_name": "InputLayer",
-        "config": {
-            "batch_input_shape": new_shape,
-            "dtype": "float32",
-            "sparse": False,
-            "name": f"{input_layer_name}",
-        },
-        "inbound_nodes": [],
-    }
-
-    model_config["layers"][1]["inbound_nodes"] = [[[f"{input_layer_name}", 0, 0, {}]]]
-    model_config["input_layers"] = [[f"{input_layer_name}", 0, 0]]
-
-    new_model: tf.keras.Model = keras_model.__class__.from_config(
+    model_config["layers"][0]["config"]["batch_input_shape"] = new_shape
+    new_model: tf.keras.Model = tf.keras.Model.from_config(
```

Comment:
Instead of recreating the entire config and specifying the `inbound_nodes` and `input_layers`, I just adjusted the `batch_input_shape`. Both pytest and my own test show that this works as expected (at least on the examples that were tested). @talmo Do you see anything problematic about this?